R/impact_method2.R

```diff
@@ -0,0 +1,442 @@
+    impact_simplified <- rbind(impact[[i]]$impact_simplified, impact_simplified)
+  }
+  ## order output
+  if(method == "method1") {
+    impact_full <- impact_full[order(impact_full$index, impact_full$country, impact_full$year, impact_full$age), ]
+    impact_simplified <- impact_simplified[order(impact_simplified$index, impact_simplified$impact_type, impact_simplified$country, impact_simplified$year), ]
+    
+  } else {
+    impact_full <- impact_full[order(impact_full$index, impact_full$support_type, impact_full$country, impact_full$year, impact_full$age), ]
+    impact_simplified <- impact_simplified[order(impact_simplified$index, impact_simplified$impact_type, impact_simplified$support_type, impact_simplified$country, impact_simplified$year), ]
+    
+  }
+  row.names(impact_full) <- NULL
+  row.names(impact_simplified) <- NULL
+  return(list(impact_full = impact_full, impact_simplified = impact_simplified))
+}
+
+line_up <- function(recipe, group, burden_outcomes) {
+  # The line_up function will subject to change, once impact recipes are imported into montagu
+  stopifnot(nrow(recipe) == 1L)
+  ### burden_outcome
+  burden_outcome <- unlist(strsplit(recipe$burden_outcome, ","))
+  j <- duplicated(burden_outcome)
+  if(any(j)) stop("Duplicated burden_outcomes not expected.")
+  
+  j <- match(burden_outcome, burden_outcomes$code)
+  k <- is.na(j)
+  if(any(k)) stop("Unexpected burden outcome in recipe file.")
+  burdens <- data.frame(index = recipe$index,
+                        burden_outcome_id = burden_outcomes$id[j])
+  
+  ### meta ids
+  ids <- c("responsibility_set_id", "responsibility_id", "burden_estimate_set_id", "stochastic_burden_estimate_set_id")
+  
+  j <- recipe$touchstone == group$touchstone &
+    recipe$modelling_group == group$modelling_group &
+    recipe$disease == group$disease &
+    recipe$baseline == group$scenario
+  v <- unique(group[j, ][ids])
+  if(nrow(v) == 0) stop("Impact recipe error: no known baseline burden_estimate_set.")
+  stopifnot(nrow(v) == 1L)
+  v$coeff <- 1
+  recipe_base <- cbind(recipe, v)
+  
+  k <- recipe$touchstone == group$touchstone &
+    recipe$modelling_group == group$modelling_group &
+    recipe$disease == group$disease &
+    recipe$focal_ingredient == group$scenario
+  w <- unique(group[k, ][ids])
+  if(nrow(w) == 0) stop("Impact recipe error: no known focal burden_estimate_set.")
+  stopifnot(nrow(w) == 1L)
+  w$coeff <- -1
+  recipe_focal <- cbind(recipe, w)
+  recipe_base$coverage_set_id <- recipe_focal$coverage_set_id
+  
+  stopifnot(v$responsibility_set_id == w$responsibility_set_id)
+  recipe <- rbind(recipe_base, recipe_focal)
+  recipe2 <- merge(recipe, burdens, by = "index")
+  recipe2
+  
+}
+
+sql_in <- function(items) {
+  paste0("(",
+         paste(items, collapse=","),
+         paste0(")"))
+}
+
+sql_in_text <- function(items) {
+  paste0("('",
+         paste(items, collapse="' ,'"),
+         paste0("')"))
+}
+
+merge_in <- function(dat, d, cols) {
```

Comment:
Also moves into `util.R` but with `.code` moved into the argument list with a default (so with `.code = ".code"` then used as `d[[.code]]` and `names(v) == .code`)

Reply:
Done.